sdk/resourcemanager/azure-resourcemanager-network/src/main/java/com/azure/resourcemanager/network/implementation/NetworkInterfaceImpl.java

```diff
@@ -563,9 +581,24 @@ protected void beforeCreating() {
                 .withNetworkSecurityGroup(new NetworkSecurityGroupInner().withId(networkSecurityGroup.id()));
         }
 
-        NicIpConfigurationImpl.ensureConfigurations(this.nicIPConfigurations.values());
+        NicIpConfigurationImpl.ensureConfigurations(this.nicIPConfigurations.values(), this.publicIPAddressDeleteOption);
 
         // Reset and update IP configs
         this.innerModel().withIpConfigurations(innersFromWrappers(this.nicIPConfigurations.values()));
     }
+
+    @Override
+    public NetworkInterfaceImpl withPublicIPAddressDeleteOptions(DeleteOptions deleteOptions) {
+        this.publicIPAddressDeleteOption = deleteOptions;
+        return this;
+    }
+
+    @Override
+    public NetworkInterfaceImpl withSpecifiedNicSpecifiedPipDeleteOption(
+        DeleteOptions deleteOptions, PublicIpAddress publicIpAddress, String ipConfigName) {
+        this.specifiedIPConfiguration(ipConfigName).withExistingPublicIpAddress(publicIpAddress);
+        this.specifiedIPConfiguration(ipConfigName).withPrivateIpVersion(publicIpAddress.version());
+        this.publicIPAddressDeleteOption = deleteOptions;
+        return this;
```

Comment:
The implementation seems to differ from the interface declaration? 

In interface, it's `Update delete option for specified public ip address of specified network interface ip config`, but here it seems to attach an existing PublicIpAddress to the NetworkInterface.